Source/WebCore/rendering/BreakLines.h

```diff
@@ -421,6 +439,39 @@ inline BreakLines::BreakClass BreakLines::classify(UChar character)
         default:
             return kWeird;
         }
+    case 0x0600 / 0x80:
+    case 0x0680 / 0x80:
+    case 0x0700 / 0x80:
+    case 0x0780 / 0x80:
+    case 0x0800 / 0x80:
+    case 0x0880 / 0x80:
+    case 0x0900 / 0x80:
+    case 0x0980 / 0x80:
+    case 0x1000 / 0x80:
+    case 0x1080 / 0x80:
+    case 0x1100 / 0x80:
+    case 0x1180 / 0x80:
+    case 0x1200 / 0x80:
+    case 0x1280 / 0x80:
+    case 0x1300 / 0x80:
+    case 0x1380 / 0x80:
+    case 0x1400 / 0x80:
+    case 0x1480 / 0x80:
+    case 0x1500 / 0x80:
+    case 0x1580 / 0x80:
+    case 0x1600 / 0x80:
+    case 0x1680 / 0x80:
+    case 0x1700 / 0x80:
+    case 0x1780 / 0x80:
+    case 0x1800 / 0x80:
+    case 0x1880 / 0x80:
+    case 0x1900 / 0x80:
+    case 0x1980 / 0x80:
+        return kWeird;
+    case 0x2000 / 0x80:
+        if (character == 0x2018 || character == 0x2019)
+            return kQU;
```

Comment:
I saw you explained that double quotation marks are sometimes categorized as open/close punctuation. Are we sure that single quotations are not also categorized that way? They are usually treated as alternatives to double quotes when the nesting level has increased, so I would expect them to behave the same.

Reply:
> I saw you explained that double quotation marks are sometimes categorized as open/close punctuation. Are we sure that single quotations are not also categorized that way?

Yes, confirmed via ICU rules. They behave differently in some specific locale cases.

Reply:
Yes, but you're sure we do not have that problem with single quotation marks in those locales?

Reply:
Yes, I mean, we ensured that only double quotes are excluded.